proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,243 @@
+# Test Plan
+
+All tests are to be included in the HLK test binary which ships with the OS.
+This test binary is only built in the OS repo and based off of the 'ExecTests'
+'ExecTests' source code in the DXC repo. There is a script in the WinTools repo
+which generates and annotates the HLK tests.
+
+There are three test categories we are concerned with:
+
+1. Implement DXIL Intrinsic tests:
+    - At the bottom of this document there is a table of all HLSL intrinsics and
+    their mapped DXIL OpCodes. We just need coverage for each DXIL OpCode. If the
+    DXIL OpCode column lists 'emulated' then this means that there are multiple
+    LLVM/DXIL Ops used to compute the intrinsic. We will need to do an audit of
+    them to confirm coverage of all vector ops.
+
+2. Implement LLVM Native operation tests:
+     - These are the test cases for the 'basic' HLSL operators listed in the HLSL
+     operators table below.
+     - We have added some test cases for these with the long vectors work, but we
+     will want to make sure we add HLK coverage for all of these.
+
+3. Standard loading and storing of long vectors
+     - Ensure we have some basic tests doing standard loading/storing of long vectors.
+
+# Vector Sizes to test
+
+I don't think there are any particularly interesting vector sizes to test. So I
```

Comment:
> I don't think there are any particularly interesting vector sizes to test.

I think we might be able to come up with some.

For example, I think vector<half, 7>, vector<half, 8> and vector<half, 9> may be interesting because they cover the 128-bit boundary that has been in place most of the time before and could be interesting for load/stores as well as how they work in groupshared.

Reply:
Thanks, I've updated to reflect these test cases.

Can either of you help point me in a direction to better understand how this could be interesting when testing load/stores and how they work in groupshared?

Reply:
Got some pointers offline for implementation details.